compressors/arithmetic_coding.py

```diff
@@ -435,3 +511,96 @@ def test_adaptive_arithmetic_coding():
         lossless_entropy_coder_test(
             encoder, decoder, freq, DATA_SIZE, encoding_optimality_precision=1e-1, seed=0
         )
+
+
+def _generate_2nd_order_markov(num_samples: int, seed: int = 0):
```

Comment:
maybe this can be a part of the `test_utils` as there are similar functions in `test_utils` already.

Reply:
moved there now